src/components/Slide.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import useSlide, { SlideContext } from '../hooks/useSlide';
+import { DeckContext } from '../hooks/useDeck';
+
+/**
+ *
+ *
+ */
+
+const Slide = props => {
+  const [state, _] = React.useContext(DeckContext);
+  const initialState = { currentSlideElement: 0 };
+  const { children, slideNo } = props;
+
+  const numberOfSlideElements =
+    typeof props.children === 'Array'
+      ? props.children.filter(x =>
+          Object.keys(x.props).includes('slide-element')
+        )
+      : 0;
+
+  const isActive = slideNo === state.currentSlide;
+
+  const SlideElements = Array.isArray(props.children)
+    ? props.children.map((x, i) => {
+        if (Object.keys(x.props).includes('slide-element')) {
+          return {
+            ...x,
+            props: {
+              ...x.props,
+              'slide-element': i,
+              style:
+                x.props['slide-element'] === state.currentSlideElement
+                  ? { backgroundColor: 'orange' }
+                  : { border: '2px solid red' }
+            }
+          };
+        }
+      })
+    : props.children;
+  return (
+    <div
+      style={
+        isActive ? { backgroundColor: 'blue' } : { border: '2px solid red' }
+      }
+    >
+      <p>{slideNo}</p>
+      <SlideContext.Provider
+        value={useSlide(initialState, isActive, numberOfSlideElements)}
```

Comment:
Should we put some logic to advance to the next slide if we're done "showing" all the slide elements? I was thinking something like,

```js
const [{  currentSlideElement }] = useSlide(initialState, isActive, numberOfSlideElements); // move this to top of component
const [ _, dispatch ] = useContext(DeckContext);

useEffect(() => {
  if (currentSlideElement >= numberOfSlideElements) {
    dispatch('next slide')
  }
  if (currentSlideElement < 0) {
    dispatch('prev slide')
  }
}, [currentSlideElement])

Reply:
This should be covered by src/hooks/useSlide.js now I think :)